client/landing/stepper/declarative-flow/internals/steps-repository/pattern-assembler/screen-category-list.tsx

```diff
@@ -0,0 +1,130 @@
+import { Button } from '@automattic/components';
+import { __experimentalNavigatorBackButton as NavigatorBackButton } from '@wordpress/components';
+import { __experimentalUseFocusOutside as useFocusOutside } from '@wordpress/compose';
+import { Icon, chevronRight } from '@wordpress/icons';
+import classNames from 'classnames';
+import { useTranslate } from 'i18n-calypso';
+import { useMemo, useState } from 'react';
+import { createPortal } from 'react-dom';
+import useCategoriesOrder from './hooks/use-categories-order';
+import NavigatorHeader from './navigator-header';
+import PatternListPanel from './pattern-list-panel';
+import { useSectionPatterns } from './patterns-data';
+import type { Pattern, Category } from './types';
+import './screen-category-list.scss';
+
+interface Props {
+	categories: Category[];
+	sectionsMapByCategory: { [ key: string ]: Pattern[] };
+	onDoneClick: () => void;
+	onSelect: (
+		type: string,
+		selectedPattern: Pattern | null,
+		selectedCategory: string | null
+	) => void;
+	replacePatternMode: boolean;
+	selectedPattern: Pattern | null;
+	wrapperRef: React.RefObject< HTMLDivElement > | null;
+}
+
+const ScreenCategoryList = ( {
+	sectionsMapByCategory,
+	categories,
+	onDoneClick,
+	replacePatternMode,
+	onSelect,
+	selectedPattern,
+	wrapperRef,
+}: Props ) => {
+	const translate = useTranslate();
+	const [ selectedCategory, setSelectedCategory ] = useState< string | null >( null );
+	const [ openPatternList, setOpenPatternList ] = useState< boolean | null >( null );
+	const sectionPatterns = useSectionPatterns();
+	const categoriesInOrder = useCategoriesOrder( categories );
+
+	const handleFocusOutside = () => {
+		setOpenPatternList( false );
+		setSelectedCategory( null );
+	};
+
+	return (
+		<div className="screen-container" { ...useFocusOutside( handleFocusOutside ) }>
```

Comment:
We need to add `tabIndex={ -1 }` to make `useFocusOutside` work as expected. Now if you click on the screen container (not on the category), the pattern list panel will be closed and it's not user-friendly. However, there is another issue when you click on the pattern list panel (not on the pattern) since it's not a child of the screen container. Let's tackle it in the follow-up PR

Reply:
`useFocusOutside` doesn't work on Safari, and when it works it doesn't do what we want, which is "Click on the large preview to close Pattern List". That's why I'm adding back the previous solution that works perfectly.